transmogrifier/sources/xml/oaidc.py

```diff
@@ -36,45 +36,34 @@ def get_optional_fields(self, xml: Tag) -> dict | None:
         # citation: uses fallback get_citation() method
 
         # content_type
-        fields["content_type"] = [self.source]
+        fields["content_type"] = self.get_content_type()
 
         # contents: not set in this transformation
 
         # contributors
-        for creator in [c for c in xml.find_all("dc:creator") if c.string]:
-            fields.setdefault("contributors", []).append(
-                timdex.Contributor(
-                    value=str(creator.string),
-                    kind="Creator",
-                )
-            )
+        fields["contributors"] = self.get_contributors(source_record) or None
 
-        # dates
-        fields["dates"] = self.get_dates(source_record_id, xml) or None
+        # [TODO] dates
+        fields["dates"] = self.get_dates(source_record_id, source_record) or None
 
         # edition: not set in this transformation
 
         # file_formats: not set in this transformation
 
         # format
-        fields["format"] = "electronic resource"
+        fields["format"] = self.get_format()
```

Comment:
Similar to [the comment on content_type](https://github.com/MITLibraries/transmogrifier/pull/177/files#r1608711727), I opted to write a `get_*()` method for consistency, even though it only returns a hardcoded string.

Reply:
I'm sure this has been hashed out in the comment above and elsewhere, but from my POV having these field methods is **key**.  When we get into orchestration, and it's not some imperative code like this where we can just set values, I'd imagine it will be looking for specifc methods to run, effectively making it the _only_ way to get values for those fields.